platforms/linux/arm.toolchain.cmake

```diff
@@ -57,6 +57,16 @@ if(NOT DEFINED CMAKE_CXX_FLAGS)
 
   set(CMAKE_CXX_FLAGS           "${CMAKE_CXX_FLAGS} -fdata-sections -Wa,--noexecstack -fsigned-char -Wno-psabi")
   set(CMAKE_C_FLAGS             "${CMAKE_C_FLAGS} -fdata-sections -Wa,--noexecstack -fsigned-char -Wno-psabi")
+  if(CMAKE_SYSTEM_PROCESSOR STREQUAL "aarch64")
+    # see https://gcc.gnu.org/onlinedocs/gcc/AArch64-Options.html
+    if(ENABLE_BF16)
+      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=armv8.6-a")
+      set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -march=armv8.6-a")
```

Comment:
Toolchain files are not used for native compilation.

We need a robust solution for such cases too, e.g. through `OPENCV_EXTRA_*` flags (currently they are applied after the feature checks)

Reply:
For native compilation we assume that toolchain have architecture flags set by default.

Reply:
For example on Armbian 12 I have:
```
g++ -march=native -Q --target-help | grep march
  -march=                     		armv8.2-a+crypto+fp16+rcpc+dotprod
```

Reply:
> `-march=native`

It is not a default actually.

"Native" case is already properly handled by the current scripts: https://github.com/opencv/opencv/blob/4.10.0/cmake/OpenCVCompilerOptimizations.cmake#L167

Reply:
For native compilation we can:
* leave it to user - set `CMAKE_CXX_FLAGS` or write your own cmake toolchain file
* make our own cmake toolchain files similar to those used for cross-compilation (feature controlling flags should be the same)

BTW, we don't handle `-mcpu=native` (equivalent to `-march=native -mtune=native` if I understand correctly). Example from Armbian:
```
orangepi@orangepi5:~$ gcc  -Q --help=target | egrep 'mtune|mcpu|march'
  -march=                     		armv8-a
  -mcpu=                      		generic
  -mtune=                     		generic
orangepi@orangepi5:~$ gcc -march=native  -Q --help=target | egrep 'mtune|mcpu|march'
  -march=                     		armv8.2-a+crypto+fp16+rcpc+dotprod
  -mcpu=                      		generic
  -mtune=                     		generic
orangepi@orangepi5:~$ gcc -mcpu=native  -Q --help=target | egrep 'mtune|mcpu|march'
  -march=                     		armv8.2-a
  -mcpu=                      		cortex-a76.cortex-a55+crypto
  -mtune=                     		cortex-a76.cortex-a55
```